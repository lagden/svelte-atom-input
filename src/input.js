/* istanbul ignore file */
import Input from './Input.svelte'

export {
	Input
}
