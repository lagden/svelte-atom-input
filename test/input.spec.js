/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {Input} from '../src/input.js'

describe('Input', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Input)
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Input, {
			props: {
				class: 'customCss',
				type: 'tel',
				required: true
			}
		})
		expect(container).toMatchSnapshot()
	})

	test('invalid type', () => {
		const {container} = render(Input, {
			props: {
				type: 'xxx'
			}
		})
		expect(container).toMatchSnapshot()
	})
})
