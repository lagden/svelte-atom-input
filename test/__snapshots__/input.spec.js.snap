// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input invalid type 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__base "
      type="text"
    />
  </div>
</body>
`;

exports[`Input should match snapshot default 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__base "
      type="text"
    />
  </div>
</body>
`;

exports[`Input should match snapshot props 1`] = `
<body>
  <div>
    <input
      class="_atom_frm__base customCss"
      required=""
      type="tel"
    />
  </div>
</body>
`;
